Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccreissuance/core/reissuer/DccReissuerTest.kt

```diff
@@ -0,0 +1,182 @@
+
+        coVerify(exactly = 0) { dccSwapper.swap(any(), any()) }
+    }
+
+    @Test
+    fun `startReissuance throws DCC_RI_NO_RELATION if no relation index`() = runBlockingTest {
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = listOf(
+                DccReissuanceResponse.DccReissuance(
+                    certificate = "HC1:6BFOXN...",
+                    relations = listOf(
+                        DccReissuanceResponse.Relation(
+                            index = 1,
+                            action = "replace"
+                        )
+                    )
+                )
+            )
+        )
+        shouldThrow<DccReissuanceException> {
+            dccReissuer().startReissuance(dccReissuanceDescriptor = certificateReissuance)
+        }.errorCode shouldBe DccReissuanceException.ErrorCode.DCC_RI_NO_RELATION
+
+        coVerify(exactly = 0) { dccSwapper.swap(any(), any()) }
+    }
+
+    @Test
+    fun `startReissuance throws DCC_RI_NO_RELATION if no relation action`() = runBlockingTest {
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = listOf(
+                DccReissuanceResponse.DccReissuance(
+                    certificate = "HC1:6BFOXN...",
+                    relations = listOf(
+                        DccReissuanceResponse.Relation(
+                            index = 1,
+                            action = "combine"
+                        )
+                    )
+                )
+            )
+        )
+        shouldThrow<DccReissuanceException> {
+            dccReissuer().startReissuance(dccReissuanceDescriptor = certificateReissuance)
+        }.errorCode shouldBe DccReissuanceException.ErrorCode.DCC_RI_NO_RELATION
+
+        coVerify(exactly = 0) { dccSwapper.swap(any(), any()) }
+    }
+
+    @Test
+    fun `startReissuance works`() = runBlockingTest {
+        val dccReissuance = DccReissuanceResponse.DccReissuance(
+            certificate = "HC1:6BFOXN...",
+            relations = listOf(
+                DccReissuanceResponse.Relation(
+                    index = 0,
+                    action = "replace"
+                )
+            )
+        )
+        coEvery { dccReissuanceProcessor.requestDccReissuance(any()) } returns DccReissuanceResponse(
+            dccReissuances = listOf(dccReissuance)
+        )
+        shouldNotThrow<DccReissuanceException> {
+            dccReissuer().startReissuance(dccReissuanceDescriptor = certificateReissuance)
+        }
+
+        coVerify(exactly = 1) {
+            dccSwapper.swap(
+                dccReissuance,
+                certificateReissuance.certificateToReissue
+            )
+        }
+    }
+
+    @Test
+    fun `startReissuance should throw what swapper throws`() = runBlockingTest {
```

Comment:
`what` => `when`

Reply:
I meant  `what` 😄 , it throws what is thrown by the swapper , propagates the error

Reply:
Ahh :) now I get it. It makes sense :)